File: CS/Example/Models/Northwind.cs
using System.Linq;
using System.Web;

namespace Example.Models {
    public static class NorthwindDataProvider {
        const string NorthwindDataContextKey = "DXNorthwindDataContext";

        public static NorthwindDataContext DB {
            get {
                if (HttpContext.Current.Items[NorthwindDataContextKey] == null)
                    HttpContext.Current.Items[NorthwindDataContextKey] = new NorthwindDataContext();
                return (NorthwindDataContext)HttpContext.Current.Items[NorthwindDataContextKey];
            }
        }

        public static IQueryable<Customer> GetCustomers() {
            return DB.Customers;
        }
        public static IQueryable<Invoice> GetInvoices() {
            return DB.Invoices;
        }
    }
}

File: VB/Example/Models/Northwind.vb
Imports System.Linq
Imports System.Web

Namespace Example.Models
	Public Module NorthwindDataProvider
		Private Const NorthwindDataContextKey As String = "DXNorthwindDataContext"

		Public ReadOnly Property DB() As NorthwindDataContext
			Get
				If HttpContext.Current.Items(NorthwindDataContextKey) Is Nothing Then
					HttpContext.Current.Items(NorthwindDataContextKey) = New NorthwindDataContext()
				End If
				Return DirectCast(HttpContext.Current.Items(NorthwindDataContextKey), NorthwindDataContext)
			End Get
		End Property

		Public Function GetCustomers() As IQueryable(Of Customer)
			Return DB.Customers
		End Function
		Public Function GetInvoices() As IQueryable(Of Invoice)
			Return DB.Invoices
		End Function
	End Module
End Namespace